sygnal/gcmpushkin.py

```diff
@@ -389,6 +417,10 @@ async def dispatch_notification(self, n, device, context):
 
             if len(pushkeys) > 0:
                 log.info("Gave up retrying reg IDs: %r", pushkeys)
+                if len(pushkeys) == total_pushkey_count:
+                    # if every single device failed, we should properly declare
+                    # this a failure
+                    raise NotificationDispatchException("Gave up retrying FCM.")
```

Comment:
is this a separate thing?

Reply:
Yes, it is, wonder how that came about, again… (memory hazy but I think it showed up during testing with proxies).

I'll take this out and rePR it if it's worthwhile.